bindgen/src/JSONBuilder.ts

```diff
@@ -101,19 +106,33 @@ export class JSONBindingsBuilder extends BaseVisitor {
     return sources.filter(hasNearDecorator);
   }
 
+  static checkTestBuild(sources: Source[]) {
+    this.isTest = sources.some((s) => s.normalizedPath.includes(".spec."));
+  }
+
   visitClassDeclaration(node: ClassDeclaration): void {
     if (!this.exportedClasses.has(toString(node.name))) {
       this.exportedClasses.set(toString(node.name), node);
     }
     super.visitClassDeclaration(node);
   }
 
-  visitFunctionDeclaration(node: FunctionDeclaration): void {
-    let isNotEntry = !isEntry(node);
+  needsWrapper(node: FunctionDeclaration): boolean {
+    let isExport = node.is(CommonFlags.EXPORT);
     let alreadyWrapped = this.wrappedFuncs.has(toString(node.name));
-    let isNotExport = !node.is(CommonFlags.EXPORT);
     let noInputOrOutput = numOfParameters(node) == 0 && returnsVoid(node);
-    if (isNotEntry || alreadyWrapped || isNotExport || noInputOrOutput) {
+    if (
+      !isExport ||
+      alreadyWrapped ||
+      noInputOrOutput ||
```

Comment:
nice! just realizing now that:

```ts
export { myFunc } from './path.ts'
```

was always possible as long as `myFunc` han to inputs or outputs.